elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/AggregationDataStore.java

```diff
@@ -35,8 +49,13 @@ public AggregationDataStore(QueryEngineFactory queryEngineFactory, MetaDataStore
      */
     @Override
     public void populateEntityDictionary(EntityDictionary dictionary) {
-        metaDataStore.loadMetaData(dictionary);
-        queryEngine = queryEngineFactory.buildQueryEngine(dictionary, metaDataStore);
+        for (Class<? extends Annotation> cls : AGGREGATION_STORE_CLASSES) {
+            ClassScanner.getAnnotatedClasses(modelPackageName, cls).stream().forEach(modelClass -> {
+                dictionary.bindEntity(modelClass);
+            });
+        }
+
+        queryEngine = queryEngineFactory.buildQueryEngine(metaDataStore);
 
         /* Add 'grain' argument to each TimeDimensionColumn */
         for (AnalyticView table : metaDataStore.getMetaData(AnalyticView.class)) {
```

Comment:
We need to ensure that the metadatastore is populated before this datastore. In current logic, the `metadatastore.loadMetadata()` can happen after aggregationDataStore populating the dictionary.

Reply:
Fixed